File: Assets/Scripts/HookScript.cs
using UnityEngine;

[System.Serializable]
public partial class HookScript : MonoBehaviour
{
    public virtual void Start()
    {
    }

    public virtual void Update()
    {
    }

    public virtual void OnTriggerEnter(Collider c)
    {
        if (c.transform.name == "ali")
        {
            GameObject g = c.gameObject;
            aliScript a = (aliScript)g.GetComponent("aliScript");
            a.Hooked = true;
            transform.parent.parent = g.transform;
        }
    }

}

File: Assets/Scripts/HookScript.js
#pragma strict

function Start () {

}

function Update () {

}




function OnTriggerEnter(c:Collider)
{
	if(c.transform.name=="ali")
	{
		var g:GameObject=c.gameObject;
		var a:aliScript=g.GetComponent("aliScript");
		
		a.Hooked=true;
		transform.parent.parent=g.transform;
	}	
}